### DISCUSSION TITLE
The image becomes greyscale

### DISCUSSION THREAD

**Discussion by steka** [ImageMagick/ImageMagick#7330] [Help] [answered]
### ImageMagick version

7.1.1-32

### Operating system, version and so on

Windows 10

### Description

[](url)I do the following commands:

```
$ magick convert rose: rose.png

$ magick convert -size 100x100 pattern:HS_BDIAGONAL pattern.png

$ magick convert pattern.png rose.png -gravity west -composite result.png

$ magick identify rose.png pattern.png result.png
rose.png PNG 70x46 70x46+0+0 8-bit sRGB 7011B 0.001u 0:00.000
pattern.png PNG 100x100 100x100+0+0 8-bit Grayscale Gray 2c 342B 0.000u 0:00.000
result.png PNG 100x100 100x100+0+0 8-bit Grayscale Gray 256c 2901B 0.000u 0:00.000

$ 
```
![screenshot](https://github.com/ImageMagick/ImageMagick/assets/2719766/b67fe364-8ba7-49b4-856c-f8196d749bdf)
Why does the result.png become greyscale, and how to avoid it?
I just what the color rose image to be surronded by the black/white pattern.

**Comment by GeeMack** [comment, marked as answer]
When IM saves an image with only black and white, it saves it as a grayscale image. Then when doing a composite, the colorspace info is taken from the base image, the black and white pattern. These commands will avoid that...

    magick rose: rose.png
    magick -size 100x100 pattern:HS_BDIAGONAL pattern.png
    magick pattern.png -colorspace srgb rose.png -gravity west -composite result.png

That adds "-colorspace srgb" in the last line to make sure the command overrides that grayscale info from the pattern image.

    magick -size 100x100 pattern:HS_BDIAGONAL rose: -gravity west -composite result.png

That command just makes the pattern then reads in the rose to composite it on top, and uses the "srgb" colorspace since the pattern hasn't been saved yet.

**Reply by steka** [reply]
Thanks, that didn't just gave an solution, but it also gave an explanation on why, and that will not just help me now, it will also help me in the future.
Thanks, and a gold star to you!
